src/main/flight/position.c

```diff
@@ -60,30 +64,12 @@ PG_RESET_TEMPLATE(positionConfig_t, positionConfig,
     .altNumSatsBaroFallback = POSITION_DEFAULT_ALT_NUM_SATS_BARO_FALLBACK,
 );
 
-static int32_t estimatedAltitudeCm = 0;                // in cm
-
-#define BARO_UPDATE_FREQUENCY_40HZ (1000 * 25)
-
 #ifdef USE_VARIO
 static int16_t estimatedVario = 0;                   // in cm/s
 
-int16_t calculateEstimatedVario(int32_t baroAlt, const uint32_t dTime) {
-    static float vel = 0;
-    static int32_t lastBaroAlt = 0;
-
-    int32_t baroVel = 0;
-
-    baroVel = (baroAlt - lastBaroAlt) * 1000000.0f / dTime;
-    lastBaroAlt = baroAlt;
-
-    baroVel = constrain(baroVel, -1500.0f, 1500.0f);
-    baroVel = applyDeadband(baroVel, 10.0f);
-
-    vel = vel * CONVERT_PARAMETER_TO_FLOAT(barometerConfig()->baro_cf_vel) + baroVel * (1.0f - CONVERT_PARAMETER_TO_FLOAT(barometerConfig()->baro_cf_vel));
-    int32_t vel_tmp = lrintf(vel);
-    vel_tmp = applyDeadband(vel_tmp, 5.0f);
-
-    return constrain(vel_tmp, SHRT_MIN, SHRT_MAX);
+int16_t calculateEstimatedVario(float baroAltVelocity) {
+    baroAltVelocity = applyDeadband(baroAltVelocity, 10.0f);
```

Comment:
(deadband is usually only improving visual appearance of graph, while actually degrading data quality)